### PR TITLE
Migrate about tabs to cms & cleanup types

### PR DESCRIPTION
**Pull request by Aerilym**
image

### CODE REVIEW

File: components/AboutSection/Layout/index.tsx [hunk 1]
```diff
@@ -1,67 +1,58 @@
...
+  tabs: Array<AboutPageTab>;
+  items?: Array<ILegals>;
 };
 
 export default function AboutLayout(props: Props) {
-  const { tab, children } = props;
+  const { page, tabs, items } = props;
+  const pageTitle = page.useExactTitle
+    ? page.title
+    : `OPTF | ${page.title} | Privacy is a fundamental right.`;
```
Comment (yougotwill) [inline components/AboutSection/Layout/index.tsx]: Could this go into the custom head as a default?

File: components/CustomHead/index.tsx [hunk 2]
```diff
@@ -1,15 +1,24 @@
...
 import { useRouter } from 'next/router';
+import { IPageMetadata } from '@/types/cms';
 
 interface Props {
   title?: string;
-  metadata?: IPageMetadata;
+  metadata?: CustomMetadata;
 }
 
+export const parseMetadata = (pageMetadata: IPageMetadata): CustomMetadata => {
```
Comment (yougotwill) [inline components/CustomHead/index.tsx]: As mentioned above maybe but that pageTitle code here as a function

File: components/RichPage/index.tsx [hunk 3]
```diff
@@ -5,16 +5,19 @@ import { IPage } from '@/types/cms';
...
 interface Props {
   page: IPage;
 }
 
 export default function RichPage(props: Props): ReactElement {
   const { page } = props;
-  const pageTitle = page ? page.title : '';
+  const pageTitle = page.useExactTitle
+    ? page.title
+    : `OPTF | ${page.title} | Privacy is a fundamental right.`;
```
Comment (yougotwill) [inline components/RichPage/index.tsx]: Yeah it's duplicated and that is no bueno 🤷

File: pages/[slug].tsx [hunk 4]
```diff
@@ -28,20 +40,25 @@ export default function Page(props: Props): ReactElement {
...
-
   try {
     const content: IPage | IPost = await fetchEntryBySlug(slug);
 
     // embedded links in content body need metadata for preview
     content.body = await generateLinkMeta(content.body);
-    const props: Props = { content };
+    const { aboutPageTabs } = await fetchSettings();
+
+    console.log(aboutPageTabs);
```
Comment (yougotwill) [inline pages/[slug].tsx]: clean up logging

File: types/cms.ts [hunk 5]
```diff
@@ -59,12 +59,19 @@ export interface IFetchBlogEntriesReturn extends IFetchEntriesReturn {
   entries: Array<IPost>;
 }
 
-export interface IPage {
+export type IPageMetadata = {
```
Comment (yougotwill) [inline types/cms.ts]: Removes the I's from the type name since it is no longer an interface

**Review by yougotwill — CHANGES_REQUESTED** [review]

**Review by yougotwill — CHANGES_REQUESTED** [review]
Ran things on the staging site and have some more feedback. 

1. Can we have more padding on the about nav between the hover underline and the dotted border please?

Screenshot 2024-05-01 at 11 00 50

2. First page load when swapping between the different about tabs feels sluggish. I'm thinking adding a nice fade-in/out animation when you change pages might be nice? Could you investigate please?

3. The email sign up UI is broken on staging. I think it's because the substack setup isn't include the staging url? Can you see if you can fix it so that one of us doesn't keep thinking it's a bug in future please 🙏

4. I think we should have a Current Report section with an Iframe on the Annual Reports like the Transparency reports? Or do you think the file is too big?

5. Funding page headings are too small. Also the email address is incorrect (not your fault) "to secretary@optf.ngp" should be "to secretary@optf.ngo"

6. On the donations page the copy button toast is missing some padding on the right side? "Copied!" toast that happens on click shown bottom right.

File: components/AboutSection/Tabs/Donations.tsx [hunk 6]
```diff
@@ -1,97 +1,25 @@
...
-            onClick={copyClipboard}
-            className="px-3 py-1 ml-3 font-bold bg-gray-200 hover:text-white hover:bg-blue-500"
-          >
-            Copy
-          </button>
-        </li>
+        {Object.entries(WALLETS).map(([key, value]) => (
+          <li key={key}>
+            <b className="whitespace-nowrap">{key} Address:</b>{' '}
+            <TextWithCopyButton setModalOpen={setModalOpen} copyText={value} />
```
Comment (yougotwill) [inline components/AboutSection/Tabs/Donations.tsx]: is this a modal? 👀

File: components/TextWithCopyButton/index.tsx [hunk 7]
```diff
@@ -0,0 +1,36 @@
+import { forwardRef } from 'react';
+
+export interface TextWithCopyButtonProps
+  extends React.ButtonHTMLAttributes<HTMLButtonElement> {
+  copyText: string;
+  setModalOpen: (value: boolean) => void;
```
Comment (yougotwill) [inline components/TextWithCopyButton/index.tsx]: again, is this a modal? 👀 😂

File: components/TextWithCopyButton/index.tsx [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+}
+
+const TextWithCopyButton = forwardRef<
```
Comment (yougotwill) [inline components/TextWithCopyButton/index.tsx:9]: I look forward when we wont need `forwardRef` in React 19

File: pages/[slug].tsx [hunk 9]
```diff
@@ -28,20 +40,24 @@ export default function Page(props: Props): ReactElement {
         otherPosts={props.otherPosts}
       />
     );
-  } else {
-    return <RichPage page={content} />;
+  } else if (aboutPageTabs.find((tab) => tab.slug === content.slug)) {
```
Comment (yougotwill) [inline pages/[slug].tsx:43]: I might be wrong on this but won't it run on every rerender? Do we need to make it a function with a useMemo or useCallback?

Comment (yougotwill) [reply pages/[slug].tsx:43]: like `isAboutPageTab`

File: services/cms.ts [hunk 10]
```diff
@@ -6,15 +6,16 @@ import {
...
-  IAuthor,
-  IFetchBlogEntriesReturn,
-  IFetchEntriesReturn,
-  IFetchPagesReturn,
-  IFigureImage,
-  ILegals,
-  IPage,
-  IPost,
-  ITagList,
+  CMSAuthor,
```
Comment (yougotwill) [inline services/cms.ts:9]: Damn you really wanted to rename these types huh 😂

**Review by yougotwill — CHANGES_REQUESTED** [review]

File: pages/[slug].tsx (hunk 9 above)
Comment (Aerilym) [reply pages/[slug].tsx:43]: I don't think this is an issue as these pages are statically generated and no longer actually change tabs but navigate to pages

Comment (yougotwill) [reply pages/[slug].tsx:43]: All good then. Resolved

**Review by yougotwill — APPROVED** [review]
LGTM! 🚢

**Pull request merged**